frontend/src/metabase/meta/memoize.js

```diff
@@ -0,0 +1,19 @@
+
+let memoized = new WeakMap();
```

Comment:
They will be GC'd when the parent metadata instance is GC'd thanks to the way [WeakMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) works.

Putting a `memoized` map on each of the root metadata instances would have the same effect.